duckduckgo/Android
# Move internal features out of app module

Task/Issue URL: https://app.asana.com/0/488551667048375/1205077183964550/f

### Description
Move the internal features out of `:app` module

Had to create a `:common-ui-internal` module to move the `ThemesPreviewInternalFeature` in there and incidentally re-org'ed the `common` modules

### Steps to test this PR
smoke test internal features.
verify internal features don't appear in play build
